packages/components/src/select-control/types.ts

```diff
@@ -70,6 +63,11 @@ export type SelectControlSingleSelectionProps = SelectControlBaseProps & {
 	 * @default false
 	 */
 	multiple?: false;
+	/**
+	 * The value of the selected option.
+	 *
+	 * If `multiple` is true, the `value` should be an array with the values of the selected options.
+	 */
```

Comment:
Does it make sense to make this more specific, so it's clear it's for a single select? TypeScript can often see if the component has `multiple` or not and can determine which description to display in the IDE properly:

![Screenshot 2024-08-05 at 16 14 25](https://github.com/user-attachments/assets/d4c7df73-7a4d-4950-9189-f4412dbf0ef1)
![Screenshot 2024-08-05 at 16 14 43](https://github.com/user-attachments/assets/f6d3a207-3ca4-4ad8-9ff8-954dbab38a05)

Reply:
I'd love to, but I've been avoiding writing different descriptions for "conditional" props, because they don't work well with the Storybook props table. They simply get concatenated when they differ:

<img width="490" alt="Concatenated description in Storybook props table" src="https://github.com/user-attachments/assets/a7ec0724-280b-4d5c-9514-ca13f53a3f81">

Reply:
That's not a great reason for the type description verbosity TBH 😞 I wish Storybook could handle that in a more clever way. Are we the first to have that problem? Seems like this is similar to what we need: https://github.com/storybookjs/storybook/issues/21281

Reply:
I don't necessarily dislike this limitation though. I think it is useful to know that a prop is conditionally polymorphic.